pkg/apiserver/rest/dualwriter.go

```diff
@@ -65,8 +65,21 @@ type DualWriter struct {
 	legacy LegacyStorage
 }
 
-// NewDualWriter returns a new DualWriter.
-func NewDualWriter(legacy LegacyStorage, storage Storage) *DualWriter {
+type DualWriterMode int
+
+const (
+	Mode1 DualWriterMode = iota
+	Mode2
+	Mode3
+	Mode4
+)
+
+var CurrentMode = Mode2
+
+// #TODO make CurrentMode customisable and specific to each entity
+
+// newDualWriter returns a new DualWriter.
+func newDualWriter(legacy LegacyStorage, storage Storage) *DualWriter {
 	return &DualWriter{
```

Comment:
```suggestion
	return &dualWriter{
	storage: storage,
```

Reply:
it seems weird for a private function to return a public type. I'd rather keep everything public if we need to export it or to keep everything private. Can you think if a scenario of why this should be public? Otherwise let's keep it private

Reply:
This makes sense. Let's keep this in mind for when we stop relying on the generic dual writer. I'm going to switch back to registering the generic DW temporarily.